Sources/TSCBasic/Process.swift

```diff
@@ -23,6 +23,34 @@ import Dispatch
 
 import _Concurrency
 
+/// Windows is case insensitive for the environment control block. This caused issues when Swift is used with 
+/// programs which are incorrectly cased (e.g. emacs).  This type Introduces an explicit wrapper type for 
+/// Windows to make the lookup case insensitive, canonicalising the name to lowercase.  This allows us to treat 
+/// `Path` and `PATH` identically (along with any other environment variable and case matching) which respects /// the Windows behaviour.
+public struct ProcessEnvironmentBlock {
+#if os(Windows)
+    public typealias Key = CaseInsensitiveString
```

Comment:
One last question: if we have `public subscript(_ key: String) -> String` written as is, do we need to have this `typealias` at all? Why not keep `CaseInsensitiveString` internal instead of `public`?

Reply:
The constraint on the initialiser would force us to make it public.  Now ... maybe if I can make the initialiser internal ... 🤔

Reply:
Bleh, we still need the type to be public as it is used in ProcessEnv.swift as the type of a public variable.